src/java/org/apache/cassandra/db/filter/RowFilter.java

```diff
@@ -1081,6 +1106,112 @@ protected Kind kind()
         }
     }
 
+
+    // VSTODO make generic?
+    public static class GeoDistanceExpression extends Expression
+    {
+        private final ByteBuffer distance;
+        private final Operator distanceOperator;
+        private final float searchRadiusMeters;
+        private final float searchRadiusDegreesSquared;
+        private final float searchLat;
+        private final float searchLong;
+
+        public GeoDistanceExpression(ColumnMetadata column, ByteBuffer point, Operator operator, ByteBuffer distance)
+        {
+            super(column, Operator.BOUNDED_ANN, point);
+            assert column.type instanceof VectorType && (operator == Operator.LTE || operator == Operator.LT);
+            this.distanceOperator = operator;
+            this.distance = distance;
+            searchRadiusMeters = FloatType.instance.compose(distance);
+            if (searchRadiusMeters <= 0)
+                throw new InvalidRequestException("GEO_DISTANCE radius must be non-negative, got " + searchRadiusMeters);
+            searchRadiusDegreesSquared = GeoUtil.maximumSquareDistanceForCorrectLatLongSimilarity(searchRadiusMeters);
+            var pointVector = TypeUtil.decomposeVector(column.type, point.duplicate());
+            // This is validated earlier in the parser because the column requires size 2, so only assert on it
+            assert pointVector.length == 2 : "GEO_DISTANCE requires search vector to have 2 dimensions.";
+            searchLat = pointVector[0];
+            searchLong = pointVector[1];
```

Comment:
Nit: my preference is for Lon as an abbreviation since Long has its own meaning in Java

Reply:
works for me